models/migrations/v163.go

```diff
@@ -0,0 +1,27 @@
+// Copyright 2020 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"xorm.io/xorm"
+)
+
+func restrictUserAndReactionColumnSize(x *xorm.Engine) error {
+	type Reaction struct {
+		ID             int64  `xorm:"pk autoincr"`
+		Type           string `xorm:"VARCHAR(32) INDEX NOT NULL"`
+		OriginalAuthor string `xorm:"VARCHAR(64) INDEX"`
+	}
+	if err := x.Sync2(new(Reaction)); err != nil {
```

Comment:
This will not work. `Sync2` will only modify columns from small length to big length but not opposite.